### ISSUE TITLE
OC CLI: Environment variable with a DOT in them.

### ISSUE THREAD

**Issue by gt-tech** [openshift/origin#8771]
I am creating a new application using "oc new-app" CLI command and need to pass environment variables, one of my environment variable happens to be of the format of a.b.c.d=value, please see exact command below.

I get an error:

```
error: environment variables must be of the form key=value: spring.cloud.consul.host=http://192.168.18.132
error: environment variables must be of the form key=value: spring.cloud.consul.port=18500

```
##### Version

```
oc version
oc v1.2.0-rc1
kubernetes v1.2.0-36-g4a3f9c5

```
##### Steps To Reproduce
1. Create any application using "oc new-app" and try to pass an environment variable that looks like spring.cloud.consul.host=http://192.168.18.132

My sample command was -
oc new-app openshift/openshift/wildfly-100-centos7~https://codehub.myorg.com/my-lab/ose-sample-idp.git#master -e env=dev -e tenant=oid -e "spring.cloud.consul.host"="http://192.168.18.132" -e "spring.cloud.consul.port"="18500" -e MAVEN_ARGS="-s -f boot-war-pom.xml -am -pl api clean package install -Popenshift -DskipTests -B"
##### Current Result

Error on environment variables with DOT in them
##### Expected Result

With enclosed in quotes, it should work?

**Comment by mfojtik** [comment]
@smarterclayton this is pretty easy to fix in our CLI, but will require fix in k8s to allow dots in env vars. Do you have any other concerns about this except the variables wont `export` in bash? Also `docker run -e foo.test=bar` works, so I don't see the reason it should not work in k8s.

**Comment by christian-posta** [comment]
Can't you just use FOO_BAR_COFFEE ? ditch the '.' characters since http://stackoverflow.com/questions/2821043/allowed-characters-in-linux-environment-variable-names

**Comment by bparees** [comment]
closing since periods are not valid in env variable identifiers, at least in bash.

**Comment by smarterclayton** [comment]
Hold on - just because they're not in bash doesn't mean they aren't valid to pass to the container.

**Comment by smarterclayton** [comment]
We distinguish between Posix and Bash - it should be valid to define env vars with periods in them via the container interfaces in Kube and the OpenShift APIs.  Some contexts we may not allow them to be set, but only if the only interface to them is bash.

**Comment by bparees** [comment]
@smarterclayton well k8s doesn't allow it, so unless we're going to get k8s to allow them, we can't fix this.

**Comment by bparees** [comment]
```
spec.template.spec.containers[0].env[0].name: Invalid value: "FOO.BAR": must be a C identifier (matching regex [A-Za-z_][A-Za-z0-9_]*): e.g. "my_name" or "MyName"
```

**Comment by smarterclayton** [comment]
Probably something we'll have to fix in Kube.

**Comment by smarterclayton** [comment]
Please spawn a Kube issue.

**Comment by bparees** [comment]
https://github.com/kubernetes/kubernetes/issues/30120

**Comment by maiconbaumx** [comment]
Any news about this issue? I'm having the same problem here.. I want to deploy an Apache Zuul from NetflixOSS, but, it appears to need a env variable with "dot", like this: origin.zuul.client.DeploymentContextBasedVipAddresses=HELLO-NETFLIX-OSS


https://github.com/Netflix-Skunkworks/zerotodocker/wiki/Running-the-Images

**Comment by bparees** [comment]
it's blocked by the upstream issue i linked above: https://github.com/kubernetes/kubernetes/issues/30120

i haven't seen any movement on that issue.

**Comment by xiaods** [comment]
out of date?

**Comment by xiaods** [comment]
it'll only be available in 1.8.

**Comment by bparees** [comment]
looks like this is resolved so we might actually be able to fix this now: https://github.com/kubernetes/kubernetes/issues/2707
/remove-lifecycle stale
/lifecycle frozen
